### PR TITLE
[ONNX] Implemented cache for MMAP handles

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Implemented cache for MMAP handles

### Tickets:
 - 116326

### CODE REVIEW

File: src/frontends/onnx/frontend/src/core/tensor.hpp [hunk 1]
```diff
@@ -108,11 +108,13 @@ class Tensor {
     };
 
     Tensor() = delete;
-    explicit Tensor(const ONNX_NAMESPACE::TensorProto& tensor, const std::string& model_dir, const bool enable_mmap)
+    explicit Tensor(const ONNX_NAMESPACE::TensorProto& tensor,
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/src/core/tensor.hpp:111]:
```suggestion
    Tensor(const ONNX_NAMESPACE::TensorProto& tensor,
```

File: src/frontends/onnx/frontend/src/core/attribute.hpp [hunk 2]
```diff
@@ -202,10 +202,10 @@ class Attribute {
     Attribute() = delete;
     explicit Attribute(const ONNX_NAMESPACE::AttributeProto& attribute_proto,
```
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/src/core/attribute.hpp:203]:
```suggestion
    Attribute(const ONNX_NAMESPACE::AttributeProto& attribute_proto,
```

File: src/frontends/onnx/frontend/src/core/tensor.hpp (hunk 1 above)
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/core/tensor.hpp]: done

File: src/frontends/onnx/frontend/src/core/attribute.hpp (hunk 2 above)
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/core/attribute.hpp]: done

**Comment by gkrivor** [comment]
@ilya-lavrenov FYI test which covers multiple constants in one file exists: external_data_two_tensors_data_in_the_same_file

**Comment by ilya-lavrenov** [comment]
> @ilya-lavrenov FYI test which covers multiple constants in one file exists: external_data_two_tensors_data_in_the_same_file

why have it worked before?

**Comment by gkrivor** [comment]
> > @ilya-lavrenov FYI test which covers multiple constants in one file exists: external_data_two_tensors_data_in_the_same_file
> 
> why has it worked before?

Why not?

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

**Pull request merged**